app/models/tag.rb

```diff
@@ -1220,6 +1220,14 @@ def indirect_bookmarks(rec=false)
   ## SEARCH #######################
   #################################
 
+  # Override enqueue_to_index method from searchable.rb
+  # These changes are quick so we want to do them synchronously,
```

Comment:
> post-processing step on the ES results

Can we use a new decorator that throws away wrangled tags from `load_from_elasticsearch` database results?

Reply:
I was thinking more about an altered version of TagQuery, but with similar effects (only returning tags from the `Tag.unwrangled` scope).

Both approaches have the disadvantage that the database vs. ES discrepancy is a lot more visible to the wranglers (if they wrangle 19 of the 20 tags on a page, they'll be left with a page with 1 tag on it, even if there's a second page full of tags to wrangle). But it's likely to be the most up-to-date.

Maybe a combo would be good? Double-check the unwrangled status in the database, but also call `reindex_document` in critical places to try to get ES up-to-date as well?

Reply:
> I was thinking more about an altered version of TagQuery, but with similar effects (only returning tags from the Tag.unwrangled scope).

You're thinking of filtering the item set early on, instead of after-the-fact (like decorators), so we wouldn't have broken pagination?

Reply:
No, the pagination would be broken either way. I'd prefer not to use a decorator because I don't think the Tags themselves have any behavior that needs to be overridden or newly defined to make the page display properly; it's just the method of retrieving the tags from the database that needs modifying.

(Of course the only way to do that with the current system is with a separate class -- rather like a Decorator, but not actually wrapping a SimpleDelegator around the results -- and either a subclass of `TagQuery` or some special casing in `TagQuery#klass`. But I think there are a couple of ways to make that more flexible.)

Reply:
From a discussion with @tickinginstant elsewhere: instead of changing how enqueue_to_index works for Tag, we can call reindex_document for places where we need things reindexed quickly, namely:

- after_update in Tag, if there are changes to "unwrangleable"
- after_save in CommonTagging (change the update_search callback)